File: Dummy/hawkshaw/library/datalayer/models/FileUploadRequest.java
package com.hawkshaw.library.datalayer.models;

import Y1.K;
import com.hawkshaw.library.datalayer.models.Command;

public final class FileUploadRequest {
    private final File file;
    private final MetaData metadata;

    public static final class File {
        private final byte[] content;

        public File(byte[] bArr) {
            K.n(bArr, "content");
            this.content = bArr;
        }

        public final byte[] getContent() {
            return this.content;
        }
    }

    public static final class MetaData {
        private final String mime;
        private final String name;
        private final String path;
        private final Command.FileRequest.Source source;

        public MetaData(String str, String str2, String str3, Command.FileRequest.Source source2) {
            K.n(str, "name");
            K.n(str2, "path");
            K.n(str3, "mime");
            K.n(source2, "source");
            this.name = str;
            this.path = str2;
            this.mime = str3;
            this.source = source2;
        }

        public final String getMime() {
            return this.mime;
        }

        public final String getName() {
            return this.name;
        }

        public final String getPath() {
            return this.path;
        }

        public final Command.FileRequest.Source getSource() {
            return this.source;
        }
    }

    public FileUploadRequest(MetaData metaData, File file2) {
        this.metadata = metaData;
        this.file = file2;
        if (metaData == null && file2 == null) {
            throw new IllegalStateException("Both MetaData and File cannot be null in FileUploadRequest");
        }
    }

    public final File getFile() {
        return this.file;
    }

    public final MetaData getMetadata() {
        return this.metadata;
    }
}

File: app/src/main/java/com/hawkshaw/library/datalayer/models/FileUploadRequest.kt
package com.hawkshaw.library.datalayer.models

import kotlinx.serialization.Serializable

/**
 * Request for uploading files, containing metadata and the actual file contents
 */
class FileUploadRequest(
    val metadata: MetaData? = null,
    val file: File? = null
) {
    init {
        require(!(metadata == null && file == null)) { 
            "Both MetaData and File cannot be null in FileUploadRequest" 
        }
    }
    
    /**
     * Metadata about the file being uploaded
     */
    data class MetaData(
        val name: String,
        val path: String,
        val mime: String,
        val source: Command.FileRequest.Source
    )
    
    /**
     * File content being uploaded
     */
    data class File(
        val content: ByteArray
    ) {
        override fun equals(other: Any?): Boolean {
            if (this === other) return true
            if (javaClass != other?.javaClass) return false

            other as File

            return content.contentEquals(other.content)
        }

        override fun hashCode(): Int {
            return content.contentHashCode()
        }
    }
} 